x-pack/legacy/plugins/apm/server/lib/transactions/distribution/get_buckets/fetcher.ts

```diff
@@ -17,18 +17,38 @@ import {
 import { rangeFilter } from '../../../helpers/range_filter';
 import { Setup } from '../../../helpers/setup_request';
 
-export function bucketFetcher(
-  serviceName: string,
-  transactionName: string,
-  transactionType: string,
-  transactionId: string,
-  traceId: string,
-  bucketSize: number,
-  setup: Setup
-) {
+export function bucketFetcher({
+  serviceName,
+  transactionName,
+  transactionType,
+  transactionId,
+  traceId,
+  bucketSize,
+  setup
+}: {
+  serviceName: string;
+  transactionName: string;
+  transactionType: string;
+  transactionId?: string;
+  traceId?: string;
+  bucketSize: number;
+  setup: Setup;
+}) {
   const { start, end, uiFiltersES, client, config } = setup;
   const bucketTargetCount = config.get<number>('xpack.apm.bucketTargetCount');
 
+  const should: Array<Record<string, any>> = [
```

Comment:
I'm not sure if we should do this. It will for instance make typing `ESFilter` more accurately hard because we use lower fidelity typings. I know you're not a fan of ternaries, but one thing I'm trying is:

```js
const traceIdFilter = traceId ? [ { term: { [TRACE_ID]: traceID } }] : [];
```

and then spread that variable directly in `body.query.bool.filter`

```js
{
	body: {
		query: {
			bool: {
				filter: {
					should: [ ...traceIdFilter ]
				}
			}
		}
	}
}
```
		
IMHO, just as easy to understand, and without unnecessary types.

Reply:
Hmm.. As you know I'm scared of long ternaries 😱 but this actually looks pretty good. Will change.

Another benefit I like with your approach is that it doesn't mutate `should`, which is a Good Thing™